libs/blocks/preflight/panels/assets.js

```diff
@@ -0,0 +1,188 @@
+import { html, signal, useEffect } from '../../../deps/htm-preact.js';
+import { createTag } from '../../../utils/utils.js';
+
+function isViewportTooSmall() {
+  return !window.matchMedia('(min-width: 1200px)').matches;
+}
+
+const imagesWithMismatch = signal([]);
+const imagesWithMatch = signal([]);
+const checksPerformed = signal(false);
+const viewportTooSmall = signal(isViewportTooSmall());
+
+const groups = [
+  { title: 'Images with dimension mismatch', imgArray: imagesWithMismatch },
+  { title: 'Images with matching dimensions', imgArray: imagesWithMatch },
+];
+
+async function checkImageDimensions() {
+  if (checksPerformed.value || viewportTooSmall.value) return;
+  checksPerformed.value = true;
+
+  const allImages = [...document.querySelectorAll('main picture img')];
+  if (!allImages.length) return;
+
+  // Force load all images
+  await Promise.all(
+    allImages.map((img) => {
+      if (!img.complete) {
+        img.setAttribute('loading', 'eager');
+        return new Promise((resolve) => {
+          img.addEventListener('load', resolve);
+        });
+      }
+      return Promise.resolve();
+    }),
+  );
+
+  // Filter the loaded images to ensure they are visible, not an icon, and not an SVG
+  const images = allImages.filter((img) => img.checkVisibility()
+    && !img.closest('.icon-area')
+    && !img.src.endsWith('.svg'));
+
+  if (!images.length) return;
+
+  const maxFullWidth = 1920;
```

Comment:
Can we define this at the top of the script?

Reply:
Yes, makes sense 👍